docs/REFERENCE.md

```diff
@@ -1,13 +1,16 @@
 # API Reference
 
-`type-safe-errors` expose only three class-like abstraction: `Ok`, `Err` and `Result`.
+`type-safe-errors` exposes three class-like abstractions: [Ok](#ok), [Err](#err), and [Result](#result).
+
+For an introduction to the type-safe-errors library and its benefits, please refer to the [README](../README.md).  For framework-specific examples, please refer to the [Framework Examples](../examples) directory.
 
 ## Ok
-`Ok` objects represents a valid result of an action.
+An `Ok` object represents a valid result of an action.
 
 ### Ok.of(...)
 
-Create new `Ok` result. Static function (can be called only on imported `Ok` namespace, not on `Ok` results).
+Creates a new `Ok` result. It is a static function, meaning it can only be called on the imported `Ok` namespace and not on `Ok` result instances.
```

Comment:
Here it would be good to also inform about supporting of promises

ex: `Ok.of(Promise.resolve(3))`

Reply:
but there is not such support. `Ok.of` always expect static value.
`Result.from(() => ...)` can expect async/sync result.

Reply:
So we can also say that they are forbidden or link it `if you want to use promises use Result.from`